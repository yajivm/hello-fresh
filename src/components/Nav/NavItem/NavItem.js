import PropTypes from 'prop-types';
import { NavTitle } from './NavItem.styles'

const NavItem = ({ title, isCurrentTabActive, to, onClick }) => {
  return (
    <NavTitle
      iscurrenttabactive={isCurrentTabActive}
      to={to}
      onClick={() => onClick(title)}
      data-testid={title}
    >
      {title}
    </NavTitle>
  )
}

NavItem.propTypes = {
  title: PropTypes.string.isRequired,
  isCurrentTabActive: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default NavItem