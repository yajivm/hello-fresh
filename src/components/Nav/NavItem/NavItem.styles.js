import styled from "styled-components";
import { Link } from "react-router-dom";

export const NavTitle = styled(Link)`
  cursor: pointer;
  padding: 10px;
  width: 100px;
  text-align: center;
  font-size: 20px;
  border-bottom: 5px solid transparent;
  text-decoration: none;
  color: #000;
  ${({ iscurrenttabactive }) => iscurrenttabactive && `
    color: rgb(6, 122, 70);
    border-color: rgb(6, 122, 70);
  `}
`;