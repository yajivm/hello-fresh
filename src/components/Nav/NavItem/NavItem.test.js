import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import NavItem from '.'

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');
  return {
    ...originalModule,
    Link: props => <div {...props}></div>,
  };
});


const mockOnClick = jest.fn();
const props = {
  title: 'ATitle',
  isCurrentTabActive: true,
  to: 'path',
  onClick: mockOnClick
}

describe('NavItem', () => {

  it('should render component', () => {
    const { container } = render(<NavItem {...props} />)

    expect(container).toMatchSnapshot();
  })

  it('should call onclick prop when on click the nav item', () => {
    render(<NavItem  {...props} />)

    expect(screen.getByTestId('ATitle')).toBeInTheDocument()

    fireEvent.click(screen.getByTestId('ATitle'))

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  })
})