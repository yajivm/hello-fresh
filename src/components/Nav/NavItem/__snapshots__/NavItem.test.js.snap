// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem should render component 1`] = `
<div>
  <div
    class="sc-bczRLJ fEEeGt"
    data-testid="ATitle"
    to="path"
  >
    ATitle
  </div>
</div>
`;
